validator_client/src/beacon_node_fallback.rs

```diff
@@ -0,0 +1,424 @@
+        } else if let Err(e) = self.is_synced(slot_clock, log).await {
+            *status = Err(e);
+        } else {
+            *status = Ok(())
+        }
+
+        *status
+    }
+
+    /// Checks if the node is reachable.
+    async fn is_online(&self, log: &Logger) -> Result<(), CandidateError> {
+        let result = self
+            .beacon_node
+            .get_node_version()
+            .await
+            .map(|body| body.data.version);
+
+        match result {
+            Ok(version) => {
+                info!(
+                    log,
+                    "Connected to beacon node";
+                    "version" => version,
+                    "endpoint" => %self.beacon_node,
+                );
+                Ok(())
+            }
+            Err(e) => {
+                warn!(
+                    log,
+                    "Offline beacon node";
+                    "error" => %e,
+                    "endpoint" => %self.beacon_node,
+                );
+                Err(CandidateError::Offline)
+            }
+        }
+    }
+
+    /// Checks if the node has the correct specification.
+    async fn is_compatible(&self, spec: &ChainSpec, log: &Logger) -> Result<(), CandidateError> {
+        let yaml_config = self
+            .beacon_node
+            .get_config_spec()
+            .await
+            .map_err(|e| {
+                error!(
+                    log,
+                    "Unable to read spec from beacon node";
+                    "error" => %e,
+                    "endpoint" => %self.beacon_node,
+                );
+                CandidateError::Offline
+            })?
+            .data;
+
+        let beacon_node_spec = yaml_config
+            .apply_to_chain_spec::<E>(&E::default_spec())
+            .ok_or_else(|| {
+                error!(
+                    log,
+                    "The minimal/mainnet spec type of the beacon node does not match the validator \
+                    client. See the --network command.";
+                    "endpoint" => %self.beacon_node,
+                );
+                CandidateError::Incompatible
+            })?;
+
+        if *spec == beacon_node_spec {
+            Ok(())
+        } else {
+            error!(
+                log,
+                "The beacon node is using a different Eth2 specification to this validator client. \
+                See the --network command.";
+                "endpoint" => %self.beacon_node,
+            );
+            Err(CandidateError::Incompatible)
+        }
+    }
+
+    /// Checks if the beacon node is synced.
+    async fn is_synced<T: SlotClock>(
+        &self,
+        slot_clock: Option<&T>,
+        log: &Logger,
+    ) -> Result<(), CandidateError> {
+        if let Some(slot_clock) = slot_clock {
+            match check_synced(&self.beacon_node, slot_clock, Some(log)).await {
+                Ok(_) => Ok(()),
+                Err(_) => Err(CandidateError::NotSynced),
```

Comment:
I think we should add a `warn` log here.

Reply:
Nice, I've cherry-picked 0e11b02ce866a05b2d82da6d316fab469ac2f471